tests/runtime/json-output

```diff
@@ -39,11 +44,14 @@ PROG BEGIN { @["bpftrace"] = hist(2); @["curl"] = hist(-1); @["curl"] = hist(0);
 EXPECT_JSON runtime/outputs/hist_multiple.json
 TIMEOUT 5
 
+# TODO this can go back to just a single line of json e.g.
+# PROG i:ms:1 { @ = hist(@n++,3); if (@n > 1023) { delete(@n); exit(); }}
+# EXPECT_JSON runtime/outputs/hist_2args.json
+# when the race-condition for map deletion is fixed
```

Comment:
`delete()` is a synchronous function, so will always complete before `exit()` is run. I think the only race condition that can happen in this test is that it doesn't run 1024 times before the `TIMEOUT` expires (say if it takes 5 seconds for bpftrace to start up for some reason). Increasing the timeout to 10 seconds like you've done might be all that's needed to fix it

Reply:
> delete() is a synchronous function, so will always complete before exit() is run. I think the only race condition that can happen in this test is that it doesn't run 1024 times before the TIMEOUT expires

This is not what I was experiencing locally. The output was including the map e.g. `{"type": "map", "data": {"@n": 3}}` - and the value of @n changed often. This was happening with a more frequent interval e.g. `i:us:100`, which is why I changed it to `ms:1`. That being said, I was still occasionally seeing the map in the output (after deletion) even if this less frequent interval, which is why I just set it to 1 - this strategy seemed to work best. So perhaps there are multiple issues at play here.

Reply:
Ah, I think another possible problem is a race condition with `@n++` if the probes fire too close together. That would explain increasing the interval having helped.

But whatever the cause of us not getting `@n = 1024`, I'd expect the data in the histogram to also be incorrect. I've not reproduced this failure myself, but #3035 does show an incomplete histogram (192 entries instead of 1024) along with `@n = 202`

How about this as a fix for the race condition:
```
BEGIN { $i = 0; while ($i < 1024) { @ = hist($i, 3); $i++; } exit(); }
```

Reply:
(I'm getting a bit off track for this PR, sorry)

Reply:
> BEGIN { $i = 0; while ($i < 1024) { @ = hist($i, 3); $i++; } exit(); }

Ah good idea! This worked very well.